packages/components/src/DataList/DataList.types.ts

```diff
@@ -310,7 +315,7 @@ export interface DataListActionsProps<T extends DataListObject> {
 }
 
 export interface DataListBulkActionProps
-  extends DataListActionProps<DataListObject> {
+  extends Omit<DataListActionProps<DataListObject>, "visible"> {
```

Comment:
Not sure if this should be a future thing to address or deal with it here. 
I wonder how it would handle bulk action when the individual action is not visible. For example bulk archiving things that include an already archived items. 
Would we want to filter out the items that should be filtered or have the consumer deal with it?

Reply:
we'd most likely tell them that we've only archived 20 out of the 21 since the 1 is already archived. It would be weird to start removing actions because of the things you've selected because you'd eventually have no actions at that point

Reply:
Sounds good

Reply:
I think maybe add something to the documentation that would specify the consumer dealing with the items when handling the bulk action that they may not want to be visible for that item

Reply:
yeah, i'll hold off on that until we solidify how we actually do batch